### ISSUE TITLE
Performance of version 5.8.0 is WAY worse than 5.7.0

### ISSUE THREAD

**Issue by Toseflo** [shedaniel/LightOverlay-Fabric#83]
I was able to use version 5.7.0 without any noticeable lag, but when I use version 5.8.0 with the same radius (24 blocks), my game lags immensely. I made a quick test - I activated LightOverlay and flew with it activated in spectator mode through the nether (I only used FabricAPI, Arcitectury, ClothConfig and ModMenu for the test).
With version 5.7.0 I got around 40 fps with short lags sometimes.
With version 5.8.0 I got 5 fps at best, the was lagging constantly and often times it dropped to 1 fps.

**Comment by shedaniel** [comment]
Just to be sure, you have cache mode on right

**Comment by Toseflo** [comment]
Thanks, activating caching fixed it. But why does it lag in 5.8.0 without caching when it didn't lag in 5.7.0 without caching?

**Comment by shedaniel** [comment]
Can you actually confirm that without caching, 5.7.0 is not laggy?

**Comment by Toseflo** [comment]
Yes just tested it again. In 5.7.0 I can even turn the radius up to 32 blocks without caching and it doesn't lag

**Comment by shedaniel** [comment]
Thanks for confirming, there are certainly a few changes I did in 5.8.0 that have affected performance.
